src/treq/test/test_multipart.py

```diff
@@ -1,12 +1,11 @@
 # Copyright (c) Twisted Matrix Laboratories.
 # See LICENSE for details.
 
-import cgi
-import sys
 from typing import cast, AnyStr
 
 from io import BytesIO
 
+from multipart import MultipartParser  # type: ignore
```

Comment:
it's ok to use multipart in the testing code, to double check the implementation... even if the code under tests would have use stdlib.

Reply:
For what it's worth, the `cgi` docs [suggest](https://docs.python.org/3.11/library/cgi.html#cgi.parse_multipart) the `multipart` package:

> Deprecated since version 3.11, will be removed in version 3.13: This function, like the rest of the [cgi](https://docs.python.org/3.11/library/cgi.html#module-cgi) module, is deprecated. It can be replaced with the functionality in the [email](https://docs.python.org/3.11/library/email.html#module-email) package (e.g. [email.message.EmailMessage](https://docs.python.org/3.11/library/email.message.html#email.message.EmailMessage)/[email.message.Message](https://docs.python.org/3.11/library/email.compat32-message.html#email.message.Message)) which implements the same MIME RFCs, or with the [multipart](https://pypi.org/project/multipart/) PyPI project.

While it suggests the `email` package too, I do not think that the `email` package is representative of real-world multipart/form-data parsers, in particular because it targets email-dialect MIME that involves nesting never seen in HTTP. (As just one example, [Django's multipart parser](https://github.com/django/django/blob/45f778eded9dff59cfdd4dce8720daf87a08cfac/django/http/multipartparser.py) doesn't support nesting. I guarantee no browser ever generates it.)

A more ideal test suite would test against a variety of real-world multipart parsers, but I don't think that'd have great ROI.